app/services/geo_concerns/delivery/geoserver.rb

```diff
@@ -64,19 +64,53 @@ def workspace
         end
 
         def datastore
-          RGeoServer::DataStore.new catalog, workspace: workspace, name: file_set.id
+          @datastore ||= RGeoServer::DataStore.new catalog, workspace: workspace,
+                                                            name: file_set.id
         end
 
-        def publish_vector
-          datastore.upload_file file_path, publish: true
+        def coveragestore
+          @coveragestore ||= RGeoServer::CoverageStore.new catalog, workspace: workspace,
+                                                                    name: file_set.id
         end
 
-        def coveragestore
-          RGeoServer::CoverageStore.new catalog, workspace: workspace, name: file_set.id
+        def base_path(path)
+          path.gsub(CurationConcerns.config.derivatives_path, '')
+        end
+
+        def persist_coveragestore
+          url = "file:///#{@config[:derivatives_path]}#{base_path(file_path)}"
+          coveragestore.url = url
+          coveragestore.enabled = 'true'
+          coveragestore.data_type = 'GeoTIFF'
+          coveragestore.save
+        end
+
+        def persist_coverage
+          coverage = RGeoServer::Coverage.new catalog, workspace: workspace,
+                                                       coverage_store: coveragestore,
+                                                       name: coveragestore.name
+          coverage.title = coveragestore.name
+          coverage.metadata_links = []
+          coverage.save
+        end
+
+        def publish_vector
+          file_dir = File.dirname(file_path)
+
+          # Delete existing shape files
+          Dir.glob("#{file_dir}/*.{shp,dbf,prj,shx}").each { |f| File.delete(f) }
```

Comment:
this may be problematic -- there's a few more file extensions that are possible for Shapefiles.